src/databricks/labs/ucx/source_code/python/python_ast.py

```diff
@@ -186,11 +187,10 @@ def root(self) -> NodeNG:
     def walk(self) -> Iterable[NodeNG]:
         yield from self._walk(self._node)
 
-    @classmethod
-    def _walk(cls, node: NodeNG) -> Iterable[NodeNG]:
+    def _walk(self, node: NodeNG) -> Iterable[NodeNG]:
         yield node
         for child in node.get_children():
-            yield from cls._walk(child)
+            yield from self._walk(child)
```

Comment:
I think this can remain a class method?

Reply:
yes, maybe, it does not make sense to me though. This is a protected method thus it should not be reached from outside the class, thus losing the benefit of calling the method without initializing the class, like `Tree._walk`. Furthermore, it is only called from the `Tree.walk` method, which is a regular method already, so why should this not be a regular method?